dev_tool/src/jumpbox.ts

```diff
@@ -0,0 +1,323 @@
+    checkAWSAccess,
+    describeInstance,
+    describeSecurityGroup,
+    getSecretsForRDS,
+    startInstance,
+    stopInstance,
+} from './aws.js'
+import { NodeSSH } from 'node-ssh'
+import { retry } from './retry.js'
+import { httpRequest } from './nodeWrappers.js'
+import { Instance } from '@aws-sdk/client-ec2'
+
+function stageForEnv(env: string): string {
+    if (env === 'dev') {
+        return 'main'
+    }
+    return env
+}
+
+// waitForJumpboxToReachState repeatedly checks the ec2 instance to be in the given state or times out
+async function waitForJumpboxToReachState(
+    instanceID: string,
+    instanceStateCode: number
+): Promise<undefined | Error> {
+    const stoppedResult = await retry(async () => {
+        try {
+            const instance = await describeInstance({
+                Filters: [
+                    {
+                        Name: 'instance-id',
+                        Values: [instanceID],
+                    },
+                ],
+            })
+            if (instance instanceof Error) {
+                console.error('Error describing instance', instance)
+                return instance
+            }
+
+            const state = instance.State?.Code
+
+            if (state === undefined) {
+                return new Error('didnt get state back for the instance')
+            }
+
+            process.stdout.write('.')
+
+            if (state === instanceStateCode) {
+                return true
+            }
+        } catch (err) {
+            return err
+        }
+
+        return false
+    }, 60 * 1000) // 60 second timeout
+    if (stoppedResult instanceof Error) {
+        return stoppedResult
+    }
+    return undefined
+}
+
+// ensureJumpboxIsRunning gets the jumpbox Instance and if it's not running it starts it
+async function ensureJumpboxIsRunning(): Promise<Instance | Error> {
+    const instance = await describeInstance({
+        Filters: [
+            {
+                Name: 'tag:mcr-vmuse',
+                Values: ['jumpbox'],
+            },
+        ],
+    })
+    if (instance instanceof Error) {
+        console.error('error fetching jumpbox', instance)
+        return instance
+    }
+
+    const jumpboxStartInstance = instance
+
+    const jumpboxStartID = jumpboxStartInstance.InstanceId
+    const jumpboxStartState = jumpboxStartInstance.State?.Code
+
+    if (!jumpboxStartID || jumpboxStartState === undefined) {
+        return new Error(
+            `AWS didn't return info we needed. id: ${jumpboxStartID} state: ${jumpboxStartState}`
+        )
+    }
+
+    if (jumpboxStartState === 16) {
+        // running state code
+        console.info('Jumpbox is running')
+        return jumpboxStartInstance
+    }
+
+    console.info('Jumpbox is not running', jumpboxStartInstance.State?.Name)
+
+    if (jumpboxStartState !== 80) {
+        console.info('Jumpbox is not stopped yet. waiting to start it')
+        // wait for it to be running
```

Comment:
I think we're waiting for it to be stopped here? Probably should just fix that comment.